File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.py
answer = []
def rotate(matrix,queries) :
    global answer
    for q in queries :
        l = []
        y1,x1,y2,x2 = q[0]-1, q[1]-1, q[2]-1, q[3]-1
        
        start = matrix[y1][x1]
        prev = matrix[y1][x1]
        tmp = -1
        current_x = x1 ; current_y = y1
        move = [1,0]
        
        while( tmp != start) :
            current_y += move[1]
            current_x += move[0]
            
            tmp = matrix[current_y][current_x]
            matrix[current_y][current_x] = prev
            prev= tmp
            
            if( current_x == x2) :
                move = [0,1]
            if( current_y == y2 ) :
                move = [-1,0]
            if( current_x == x1) :
                move = [0,-1]
                
            l.append(tmp)
        answer.append(min(l))
    
def solution(rows, columns, queries):
    matrix = []
    tmp = []
    for i in range(1, (rows * columns +1) ) :
        tmp.append(i)
        if( len(tmp) == columns ):
            matrix.append(tmp)
            tmp = []
        
    rotate(matrix,queries)
        
    return answer

File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.js
function solution(rows, columns, queries) {
  const map = Array.from({ length: rows }, (_, i) => {
    return Array.from({ length: columns }, (__, j) => i * columns + j + 1);
  });
  const answer = [];

  queries.forEach(([sx, sy, tx, ty]) => {
    [sx, sy, tx, ty] = [sx - 1, sy - 1, tx - 1, ty - 1];
    let x = sx,
      y = sy;
    let tmp = map[x][y];
    let min = Number.MAX_SAFE_INTEGER;
    while (true) {
      min = min > tmp ? tmp : min;
      if (x === sx && y < ty) {
        [map[x][y + 1], tmp] = [tmp, map[x][y + 1]];
        y++;
      } else if (y === ty && x < tx) {
        [map[x + 1][y], tmp] = [tmp, map[x + 1][y]];
        x++;
      } else if (x === tx && y > sy) {
        [map[x][y - 1], tmp] = [tmp, map[x][y - 1]];
        y--;
      } else if (y === sy && x > sx) {
        [map[x - 1][y], tmp] = [tmp, map[x - 1][y]];
        x--;
        if (y === sy && x === sx) break;
      }
    }
    answer.push(min);
  });
  return answer;
}